libevmasm/RuleList.h

```diff
@@ -390,6 +390,27 @@ std::vector<SimplificationRule<Pattern>> simplificationRuleListPart7(
 		false
 	});
 
+	{
+		SimplificationRule<Pattern> pat = {
+			// A & ((2^256-1) >> B) == ((2^256-1) >> B)
+			{Instruction::AND, {A, {Instruction::SHR, {B, X}}}},
+			[=]() -> Pattern { return {Instruction::SHR, {B, X}}; },
+			false,
+			[=]() {
+				if (B.d() > 256)
+					return false;
+				unsigned bAsUint = static_cast<unsigned>(B.d());
+				return (A.d() & (u256(-1) >> bAsUint)) == (u256(-1) >> bAsUint);
+			}
+		};
+
+		rules.push_back(pat);
+
+		// ((2^256-1) >> B) & A == ((2^256-1) >> B)
+		pat.pattern = {Instruction::AND, {{Instruction::SHR, {B, X}}, A}};
```

Comment:
I think it is easier to read if we fully specify the rule here again. We could extract the condition into a local variable of function type, though.